app/models/piece.rb

```diff
@@ -63,10 +63,13 @@ def legal_diag_move?(row, col)
   def own_piece?
     #(I am totally not sure on this, but I know it needs to be your own piece)
     player_id == player.turn or game.turn #(or player_turn or something like this)
+    piece.color != @current_player.color
   end
 
   def nil_move?
     row_position == row && col_position = col
   end
 
+
```

Comment:
valid move in here
move on board
nil 
guard clause
return false unless legal move and give it row, col